controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -12,37 +13,42 @@ import (
 )
 
 func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
```

Comment:
i think in the case of `nil` VS `Removed`, the only difference is the message sends to user.
 they may not be aware of  what `nil`/'not set' really means here
Not every user will go and read API to understand the usage and always ask for clarification in "documentation" or some logs info. 

another part is:
i think nil == `Removed`, is this correct?
from "suggestion" code, it feels like nil == `Unmanaged`?  from `serviceMeshManagementState := operatorv1.Unmanaged`

Reply:
> i think in the case of nil VS Removed, the only difference is the message sends to user.

I would assume `nil` == not existing so `Removed`. Also I would expect removing this part from the .spec in YAML is the same as marking as `Removed`. So I am not sure it would be surprising for the user. Sounds quite natural **to me** (but I might be biased).

> i think nil == Removed, is this correct?

YES! I was half awake. Corrected. I meant `Removed` of course :)

Reply:
I did some updates based on the correct suggestion :wink: 
but also prints "nil" case in the message.